xdsl/printer.py

```diff
@@ -254,8 +261,13 @@ def print_block(self, block: Block, print_block_args: bool = True) -> None:
 
         self._indent += 1
         for op in block.ops:
-            self._print_new_line()
-            self.print_op(op)
+            if not (
+                op is block.last_op
+                and op.has_trait(IsTerminator)
+                and not print_block_terminator
+            ):
+                self._print_new_line()
```

Comment:
Should the printer check again that the last operation is a Terminator? It feels like doing too much to me at a first skim, but I didn't think it through.

Reply:
Yeah, not sure; are we expected to print both valid and invalid IR?
I'm leaning towards leaving it as I had written initially:

```suggestion
            if not op.has_trait(IsTerminator or print_block_terminator:
                self._print_new_line()
```